src/aten/ReduceOps.cpp

```diff
@@ -468,14 +469,70 @@ Tensor& XPUNativeFunctions::argmax_out(
     c10::optional<int64_t> dim,
     bool keepdim,
     Tensor& out) {
-  std::optional<Device> common_device = std::nullopt;
-  c10::impl::check_and_update_common_device(
-      common_device, out, "XPUNativeFunctions::argmax_out", "out");
-  c10::impl::check_and_update_common_device(
-      common_device, self, "XPUNativeFunctions::argmax_out", "self");
   out = argmax_meta(self, dim, keepdim, out);
   argmax_argmin_impl(self, dim, keepdim, out, native::xpu::argmax_kernel);
   return out;
 }
 
+Tensor XPUNativeFunctions::argmax(
+    const Tensor& self,
+    c10::optional<int64_t> dim,
+    bool keepdim) {
+  Tensor out;
+  out = argmax_meta(self, dim, keepdim, out);
+  argmax_argmin_impl(self, dim, keepdim, out, native::xpu::argmax_kernel);
+  return out;
+}
+
+void amax_impl(
+    Tensor& result,
+    const Tensor& input,
+    IntArrayRef dim,
+    bool keepdim) {
+  auto iter = native::make_reduction(
+      "amax", result, input, dim, keepdim, input.scalar_type());
+  native::xpu::max_all_launch_kernel(iter);
+}
+
+static void check_amin_amax(
+    Tensor& result,
+    const char* name,
+    const Tensor& self,
+    IntArrayRef dim,
+    bool keepdim) {
+  if (result.defined()) {
+    TORCH_CHECK(
+        self.scalar_type() == result.scalar_type(),
+        name,
+        " got illegal dtype for self, and out:",
+        self.scalar_type(),
+        result.scalar_type());
+  }
+  if (self.numel() == 0) {
+    native::zero_numel_check_dims(self, dim, name);
+  }
+}
+
+Tensor& XPUNativeFunctions::amax_out(
+    const Tensor& self,
+    IntArrayRef dim,
+    bool keepdim,
+    Tensor& out) {
+  check_amin_amax(out, "amax()", self, dim, keepdim);
+  out = xpu::resize_reduction(out, self, dim, keepdim, self.scalar_type());
```

Comment:
1. Please reuse PyTorch implementation, https://github.com/pytorch/pytorch/blob/648625b230e8e6e7478fb219ff4f0aa6a45070f5/aten/src/ATen/native/ReduceOpsUtils.h#L361
2. Lack of meta function. Output tensor data type could not be equal to self's data type. https://github.com/pytorch/pytorch/blob/648625b230e8e6e7478fb219ff4f0aa6a45070f5/aten/src/ATen/native/ReduceOps.cpp#L387